### PR TITLE
cljsc warnings, update deps & test runners

### PR DESCRIPTION
**Pull request by pkpkpk**
clears some warnings in cljs, added testing one-liners to deps.edn

### CODE REVIEW

**Review by whilo — APPROVED** [review]
Thanks a lot!

**Comment by TimoKramer** [comment]
Hi @pkpkpk. Thanks for this PR. Were you able to run the ./bin/run-cljstests? It does not work for me.

File: deps.edn [hunk 1]
```diff
@@ -1,14 +1,23 @@
 {:paths   ["src"]
- :deps    {org.clojure/clojure       {:mvn/version "1.10.3"}
-           org.clojure/clojurescript {:mvn/version "1.11.4"}
+ :deps    {org.clojure/clojure       {:mvn/version "1.11.1"}
+           org.clojure/clojurescript {:mvn/version "1.11.60"}
            org.clojure/core.async    {:mvn/version "1.5.648"}}
  :aliases {:cljs   {:extra-deps  {thheller/shadow-cljs {:mvn/version "2.17.2"}
```
Comment (TimoKramer) [inline deps.edn:5]: I think the version of shadow-cljs has to match the clojurescript version. Can you please update all the deps at once? I am using [neil ](https://github.com/babashka/neil) for it, it's awesome.

File: test/superv/node_runner.cljs [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  ([opts]
+   (cljs.test/run-tests (merge (cljs.test/empty-env) opts)
+                        'superv.async-test)))
+
+(defn -main [& args]
+  (cljs.nodejs/enable-util-print!)
+  (defmethod cljs.test/report [:cljs.test/default :end-run-tests] [_] (js/process.exit 0))
+  (run-tests))
+
+(set! *main-cli-fn* -main)
```
Comment (TimoKramer) [inline test/superv/node_runner.cljs:17]: Do you think it is necessary to have this? In theory node tests should run with `npx shadow-cljs compile node-test`. I am not really familiar with cljs so I am just asking.

File: deps.edn [hunk 3]
```diff
@@ -1,14 +1,23 @@
...
+                                  io.github.cognitect-labs/test-runner {:git/tag "v0.5.1"
+                                                                        :git/sha "dfb30dd"}}
                     :extra-paths ["test"]}
+           :jvm-test {:extra-deps  {io.github.cognitect-labs/test-runner {:git/tag "v0.5.1"
+                                                                          :git/sha "dfb30dd"}}
+                      :extra-paths ["test"]
+                      :main-opts ["-m" "cognitect.test-runner"]
+                      :exec-args {:nses [superv.async-test]
+                                  :dirs ["test"]}
+                      :exec-fn cognitect.test-runner.api/test}
```
Comment (TimoKramer) [inline deps.edn:18]: Did you see the `bin/run-unittests` script? We are using these to have a simplified CI. Does that script do what you are trying to accomplish? If not we would need to run these tests here in the script as well.

File: deps.edn [hunk 4]
```diff
@@ -1,14 +1,23 @@
...
+           :cljs-test {:extra-paths ["test"]
+                       :main-opts ["-m" "cljs.main" "-t" "node" "-m" "superv.node-runner"]}
```
Comment (TimoKramer) [inline deps.edn:20]: Did you see the `bin/run-cljstests` script? I know it is not running the node-tests atm. But when you uncomment https://github.com/replikativ/superv.async/blob/main/test/superv/async_test.cljc#L332 you should be able to run the node-tests from the script again and this part here would not be necessary.

**Review by TimoKramer — CHANGES_REQUESTED** [review]
Thanks for putting in the effort. I don't know why the test-supervisor test is failing in cljs but maybe you can find the issue ;) 👍

File: deps.edn (hunk 4 above)
Comment (TimoKramer) [reply deps.edn:20]: Then our pipeline for superv.async looks better again. That would be ace!

**Comment by pkpkpk** [comment]
I will take a deeper look at this in a week or so I am afk for now. I have no knowledge of shadow cljs, I use the compiler manually. I will change it how you see fit, was just trying clear the warnings 👍

**Review by TimoKramer — APPROVED** [review]

**Pull request merged**